packages/axelar-wasm-std/src/operators.rs

```diff
@@ -1,18 +1,47 @@
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::{to_vec, HexBinary, Uint256};
+use cosmwasm_std::{to_binary, HexBinary, Uint256};
 
 use sha3::{Digest, Keccak256};
 
+pub type OperatorsHash = [u8; 32];
+
 #[cw_serde]
 pub struct Operators {
     pub weights_by_addresses: Vec<(HexBinary, Uint256)>,
     pub threshold: Uint256,
 }
 
 impl Operators {
-    pub fn hash(&self) -> Vec<u8> {
-        Keccak256::digest(to_vec(self).expect("could not convert Operators to vec"))
-            .as_slice()
-            .into()
+    pub fn hash_id(&self) -> OperatorsHash {
+        let mut hasher = Keccak256::new();
+        hasher.update(
+            to_binary(&self.weights_by_addresses).expect("could not serialize serializable object"),
```

Comment:
Should we make sure that `weights_by_addresses` is sorted before hashing?

Reply:
yeah that seems like a good use of a constructor